errors/bad_request_error.go

```diff
@@ -8,18 +8,23 @@ import (
 
 type BadRequestError struct {
 	ApiError
-	Details []string
+
+	// Details provides specific messages and information to detail the application error
+	Details map[string]string
```

Comment:
I'm curious how the details should be provided if all I have to return is a basic message. What should be used as a key? This is just a thought and not a request to change. I'm on board but would like to see what some of the errors look like.

Reply:
I've been using "message" as the key for just a string.